CARDS-1842: Configure the periodic task to export survey data as csv

sdumitriu opened the pull request:
To test:

- Build the **CARDS-1842** branch
- Start in prems mode
- On `http://localhost:8080/system/console/configMgr` make sure there are 3 entries for `Survey csv export`
- On `http://localhost:8080/system/console/status-slingscheduler` check that the 3 CSV jobs are scheduled for next Saturday at 02:00
- Stop, checkout and build the **CARDS-1842-test** branch
- Start with `NIGHTLY_NOTIFICATIONS_SCHEDULE="0 * * * * ? *" ./start_cards.sh -P prems --dev`
- Create and fill in a Patient Information form, including name and a valid email address
- Create and fill in a Visit Information form, for UHN-ED, set 7 days ago
- Fill in and submit the the OED form
- Repeat for another patient and another UHN-ED form set 2 days ago
- Repeat for another patient and another UHN-ED form set 7 days ago, but **do not submit** this survey
- Wait 2 minutes
- Using composum at `http://localhost:8080/bin/browser.html/Forms`, **checkout** and change the `jcr:lastModified` date to yesterday for every form
- Wait 1 minute
- On disk, in the directory where cards was started from, check that:
    - `Questionnaires_OED_2023-02-08_2023-02-14_labels.csv` has the contents of the 2 submitted forms with labels for the answers and both headers, but not the unsubmitted form
    - `Questionnaires_OED_2023-02-08_2023-02-14.csv` has the contents of the 2 submitted forms with raw numbers for the answers and both headers, but not the unsubmitted form
    - `Questionnaires_Survey events_2023-02-08_2023-02-14.csv` has both survey from 7 days ago, but not the one from 2 days ago

sdumitriu committed:
CARDS-2040: As an admin I can configure the json processors and parameters that should alter the data before exporting as CSV
Added support for configuring the input selectors and the output file name

sdumitriu committed:
CARDS-2040: As an admin I can configure the json processors and parameters that should alter the data before exporting as CSV
Added support for specifying the export format, CSV or TSV

sdumitriu committed:
CARDS-1842: Configure the periodic task to export survey data as csv
CARDS-2028: Periodic task that exports Survey events forms including the dates when emails were sent and responses received
Added configurations for the scheduled exports

marta- commented on modules/scheduled-csv-export/src/main/java/io/uhndata/cards/scheduledcsvexport/ExportConfigDefinition.java:
```diff
@@ -35,6 +33,9 @@
     String SAVE_PATH = ".";
 
+    /** Default value for the file name format. */
+    String FILE_NAME_FORMAT = "ExportedForms_{questionnaire}_{date}_{time}.csv";
```
Why change the filename format? I've been showing samples left and right with the original file name.

sdumitriu replied:
There is one actual change: it used to be `ExportedForms__` but now it's just `ExportedForms_` (one underscore removed). The change from `{period}` to `{date}_{time}` is only in the default, but the actual configurations still use `{period}` in the filename format. If you think the default should be using `{period}` as well, I can change that.

marta- replied:
Never mind, I see it's properly overridden in the configuration.

sdumitriu replied:
Oh, I actually removed the `ExportedForms_` prefix from the filenames in the configuration. Should I add it back?

marta- replied:
Nope, the config looks good.

marta- commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -179,6 +179,48 @@
+      "questionnaires.to.be.exported": [
+        "/Questionnaires/Survey events"
+      ],
+      "selectors": ".dataFilter:statusNot=INCOMPLETE.csvHeader:raw",
```
No need for raw header here

marta- commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -179,6 +179,48 @@
+      "frequency.in.days": 7,
+      "export.schedule": "0 0 2 ? * SAT *",
+      "questionnaires.to.be.exported": [
+        "/Questionnaires/CPESIC",
```
CPESIC needs the column `/Questionnaires/CPESIC/OverallRatingofHospital/cpesic_hospital_overall` to be excluded. Should it be removed from this config (and the next one) and handled in 2 separate configurations?

marta- commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -179,6 +179,48 @@
+      ],
+      "selectors": ".dataFilter:statusNot=INCOMPLETE.csvHeader:raw",
+      "save.path": ".",
+      "file.name.format": "SurveyStatuses_{period}.csv",
```
Why not `Survey events`? Should we be worried about the space in the file name?

sdumitriu replied:
Could be that, no real reason for the different name, I just feel like "status" is more appropriate than "events".

sdumitriu committed:
CARDS-1842: Configure the periodic task to export survey data as csv
Exclude cpesic_hospital_overall
CARDS-2028: Periodic task that exports Survey events forms including the dates when emails were sent and responses received
No need for raw header on survey events

marta- commented:
Looks like this PR includes https://phenotips.atlassian.net/browse/CARDS-2028 and https://phenotips.atlassian.net/browse/CARDS-2040.

sdumitriu committed:
CARDS-2028: Periodic task that exports Survey events forms including the dates when emails were sent and responses received
Use the same file name format as the other CSV exports

sashaandjic reviewed (approved):
tested - all good
![image](https://user-images.githubusercontent.com/23243244/219206390-53f83cb9-1b08-4a6b-8f2c-10d702da736c.png)

![image](https://user-images.githubusercontent.com/23243244/219206349-838d101f-8e64-4028-b8f8-0dbbd97e1f61.png)

![image](https://user-images.githubusercontent.com/23243244/219206317-777e4524-b048-4856-a181-a61e1901711e.png)

![image](https://user-images.githubusercontent.com/23243244/219206278-0bb420fc-6163-40e4-8a97-0659c877877c.png)

![image](https://user-images.githubusercontent.com/23243244/219206203-81740d58-cf9f-4eca-810d-df4bf6ee8b8b.png)
